tests/conftest.py

```diff
@@ -81,34 +84,62 @@ def stats(self):
 
     def iter(self, **kwargs):
         start = kwargs.get("start", 0)
+        count = kwargs.get("count", None)
+        counter = 0
         if start:
             start = int(start.split("/")[-1])
-        count = kwargs.get("count", len(self.items) - start)
 
-        # Scrapinghub API returns all posible items even if `count` greater than possible
-        if start + count > len(self.items):
-            limit = len(self.items)
+        if kwargs.get("filter"):
```

Comment:
Maybe rewrite to only one `for` statement.
```
filter = lambda x: True
if kwargs.get('filter'):
    filter = lambda x: x.get(field_name) == value

for item in self.items...
    if counter == count: return
    elif filter(item):...

```

Reply:
I did this but then got this warning about assigning lambdas- https://www.python.org/dev/peps/pep-0008/#programming-recommendations

So I rewrote it as a function.